jenkins/Pipeline.gy

```diff
@@ -55,10 +55,9 @@ pipeline {
         stage('Deploy server and run functional tests') {
             steps {
                 sh 'envsubst < server/pbenchinacan/pod-nodata.yml | podman play kube -'
-                sleep 30
                 // Use curl to check when server is responding to API requests
-                retry(count: 10) {
-                    sleep 10
+                retry(count: 600) {
+                    sleep 1
                     sh 'curl --silent localhost:8080/api/v1/endpoints >/dev/null'
                 }
                 sh 'EXTRA_PODMAN_SWITCHES="--network host" jenkins/run server/exec_functests http://localhost:8080'
```

Comment:
Something odd is going on.  Here's the log from this snippet:
```
[Pipeline] retry
[Pipeline] {
[Pipeline] sleep
Sleeping for 1 sec
[Pipeline] sh
+ curl --silent localhost:8080/api/v1/endpoints
[Pipeline] }
[Pipeline] // retry
[Pipeline] sh
+ EXTRA_PODMAN_SWITCHES='--network host'
+ jenkins/run server/exec_functests http://localhost:8080/
```
Note that there appears to have been exactly one retry, which suggests that the server was addressable after _1 second_.  In the past, it has taken about 36 seconds.  Can you explain how the server now appears to be "up" before our first check??

Reply:
See the changes I'd like to land from PR #3081, https://github.com/distributed-system-analysis/pbench/pull/3081/files#diff-867862313b8a604ad75180306121a3544e650c5850bfbf0e371c4bf431339b1eR51.

Reply:
I am not following.

The functional tests ran without a problem, but we don't trust it so we want to add the timeouts back so that we'll feel good about it.

I am not interested in that.

I'd rather move forward, not backward.

Reply:
> Can we move forward and not backward?

Yes...in #3081, not in this PR.

The existing code, prior to this change, waits for more than 36 seconds, because that is how long it takes for the canned server to become functional; with this change, the wait is almost completely removed, and the functional tests work basically by coincidence, apparently because it takes them at least 36 seconds before they try to access the Pbench Server.

The changes that you've implemented in #3081 include a series of checks which ensure that Server is properly functional before starting the tests.  And, those changes properly include removing the 30 second pause, so there is no need to remove it from here, and doing so is premature.

This has nothing to do with trust or feeling good.  This is about moving the _`main` branch_ forward and not backward.

Reply:
> The existing code, prior to this change, waits for more than 36 seconds, because that is how long it takes for the canned server to become functional; with this change, the wait is almost completely removed, and the functional tests work basically by coincidence, apparently because it takes them at least 36 seconds before they try to access the Pbench Server.

How have you established that?

Have you re-run the tests for this PR to show that they fail some times?

I just don't see the need to revert.  If they become flaky again, we have a PR coming to fix that.  Why add more time to an already very long CI job?

Reply:
Well, my argument is silly.  I'll just drop add the `sleep 30` back and deal with this in another PR.

Reply:
Thanks, Peter.  And, FWIW, you've already dealt with this in #3081, and reverting the change here will actually save you a merge conflict (or make it smaller) there.